feat(ci): push release assets to Azure and GCS

bacongobbler opened the pull request:
This PR installs the Azure CLI and pushes release assets to Azure, making Helm 2 releases available at https://get.helm.sh.

related to #5663 

This PR differs from #5694 because Helm 2 relies on `golang:1.12.5`, which runs as root so no use of `sudo` is required. We also continue publishing to GCS to continue supporting backwards compatibility until that bucket is shut down. At that point we can remove the GCS steps.

Commit:
feat(ci): push release assets to Azure and GCS

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

hickeyma reviewed (changes requested):
Looks good @bacongobbler. One small thing [release notes](https://github.com/helm/helm/blob/master/docs/release_checklist.md#8-write-the-release-notes) need updating in release checklist.

bacongobbler commented:
Thanks! Fixed those up. I also found that `make fetch-dist` still pointed at GCS so I fixed that up as well.

Once the Helm 2.14.0 release assets are available on get.helm.sh, this should be good to merge. I'll give it a quick manual test to make sure.

bacongobbler commented:
Also applied to https://github.com/helm/helm/pull/5841 to this PR.
